src/User/GlobalUser.php

```diff
@@ -85,7 +86,7 @@ public function getLastAccessedTime()
     public function __get(string $name)
     {
         return match ($name) {
-            'roles' => $this->getRoles(),
+            'roles' => array_combine($this->getRoles(), array_map([Role::class, 'load'], $this->getRoles())),
             'name' => $this->getAccountName(),
             default => null,
```

Comment:
In D7 `roles` was an array of role names, correct? So we shouldn't be loading the roles.

Reply:
I added a stringable role class, so treating roles as strings returns the role names.

Reply:
But, it breaks existing `in_array` checks, which might be common in D7 code, to see if the user has a role (versus having a permission)

Reply:
Didn't test that. I assumed this function wasn't used for that.